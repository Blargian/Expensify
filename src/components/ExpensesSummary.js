import React from 'react';
import selectExpensesTotal from '../selectors/expenses-total';
import { connect } from 'react-redux';
import numeral from 'numeral';

export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
    const expenseWord = expenseCount===1 ? 'expense' : 'expenses';
    return(
    <div>
        <h3>Viewing {expenseCount} {expenseWord} totalling {expensesTotal}</h3>
    </div>
    )
    };

const mapStateToProps = (state) => ({
    expenseCount: state.expenses.length,
    expensesTotal: numeral(selectExpensesTotal(state.expenses)/100).format('$0,0.00'),
});

export default connect(mapStateToProps)(ExpensesSummary);