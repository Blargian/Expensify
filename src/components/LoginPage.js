import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faGoogle } from '@fortawesome/free-brands-svg-icons'

export const LoginPage = ({startLogin}) => {
        return (
            <div className="box-layout">
                <div className="box-layout__box">
                    <h1 className=".box-layout__title">Welcome to Expense Tracker</h1>
                    <p>The first step to financial freedom</p>
                    {/* <div className="box-layout__image"></div> */}
                    <button className="login-button" onClick={startLogin}><FontAwesomeIcon className="google-icon" icon={faGoogle} />Login</button>
                </div>
            </div>
        );
    };

    const mapDispatchToProps = (dispatch) => ({
        startLogin: () => dispatch(startLogin())
    });

export default connect(undefined, mapDispatchToProps)(LoginPage);