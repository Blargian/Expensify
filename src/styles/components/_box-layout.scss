.box-layout{
    align-items: center;
    background: url('/images/bg.jpg');
    background-size:cover;
    display:flex;
    height: 100vh;
    width: 100vw;
    justify-content: center;

    &:before {
    content: '';
        position: absolute;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
        background-image: linear-gradient(to bottom,#656666,#4b4a4a);
        opacity: .6; 
  }
}


.box-layout__box{
    background: fade-out(white, .15);
    border-radius: 2em;
    padding: $l-size $m-size;
    text-align: center;
    width: 25rem;
    z-index: 1;
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
}