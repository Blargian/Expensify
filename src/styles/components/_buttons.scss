.login-button{
    background-color: $blue;
    color: white;
    border: none;
    padding: $s-size;
    width: 15rem;
    font-weight: bold;
    font-size: $font-size-large;

    .google-icon{
        margin-right: 1rem;
    }
}

.button {
    background-color: $blue;
    color: white;
    border: none;
    display:inline-block;
    padding: $s-size;
    line-height: 1;
    font-size: $font-size-large;
    text-decoration: none;
}

.button-link {
    color: white;
    background: none;
    border:none;
}