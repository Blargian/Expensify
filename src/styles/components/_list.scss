.list-header {
    background-color: $light-grey;
    border: 1px solid darken($light-grey, 20%);
    display:flex;
    justify-content: space-between;
    color:$dark-grey;
    padding: $s-size $m-size;
}

.list-item {
    border: 1px solid darken($light-grey, 20%);
    border-top:none;
    color: $dark-grey;
    display: flex;
    flex-direction: column;
    padding: $s-size;
    text-decoration: none;
    transition: background .3s ease;
    &:hover {
        background: $light-grey;
    }

    @media (min-width: $desktop-breakpoint){
        align-items:center;
        flex-direction: row;
        justify-content: space-between;
        padding: $m-size
    }
}

.list-item__title {
    margin: 0;
    word-break: break-all;
}

.list-item__subtitle {
    color: $dark-grey;
    font-size: 1.4rem;
    margin: 0;
    word-break: break-all;
}

.list-item__amount {
    margin: $s-size 0 0 0;

    @media (min-width: $desktop-breakpoint) {
        margin: 0;
        padding-left: $s-size;
    }
}

.list-item--message {
    align-items: center;
    justify-content: center;
    color: $dark-grey;
    padding: $m-size;


}

.list-body{
    margin-bottom: $m-size;
    @media(min-width: $desktop-breakpoint){
        margin-bottom: $l-size;
    }
}