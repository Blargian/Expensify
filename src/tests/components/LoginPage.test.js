import React from 'react';
import {LoginPage} from '../../components/LoginPage';
import { shallow } from 'enzyme';

test('Should render the login page correctly',() => {
    const wrapper = shallow(<LoginPage/>);
    expect(wrapper).toMatchSnapshot();
});

test('Should call startLogout on button click', () => {
    const startLoginSpy = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLoginSpy}/>);
    wrapper.find('button').simulate('click');
    expect(startLoginSpy).toHaveBeenCalled();
  })