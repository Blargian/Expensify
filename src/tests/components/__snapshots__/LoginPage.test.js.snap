// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the login page correctly 1`] = `
<div>
  <button>
    Login
  </button>
</div>
`;
